src/ALE/MOM_ALE.F90

```diff
@@ -1125,8 +1218,50 @@ subroutine ALE_remap_velocities(CS, G, GV, h_old_u, h_old_v, h_new_u, h_new_v, u
       v_src(k) = v(i,J,k)
     enddo
 
+    if (CS%id_remap_delta_integ_v2>0) then
+      v2h_tot = 0.
+      do k=1,nz
+        v2h_tot = v2h_tot - h1(k) * (v_src(k)**2)
+      enddo
+    endif
+
     call remapping_core_h(CS%vel_remapCS, nz, h1, v_src, nz, h2, v_tgt)
 
+    if (variance_option .and. CS%conserve_ke) then
+    ! Conserve ke_v by correcting baroclinic component.
+    ! Assumes total depth doesn't change during remap, and
+    ! that \int v(z) dz doesn't change during remap.
+      ! First get barotropic component
+      v_bt = 0.0
+      do k=1,nz
+        v_bt = v_bt + h2(k) * v_tgt(k) ! Dimensions [H L T-1]
+      enddo
+      v_bt = v_bt / (sum(h2(1:nz)) + GV%H_subroundoff) ! Dimensions return to [L T-1]
+      ! Next get baroclinic ke = \int (u-u_bt)^2 from source and target
+      ke_c_src = 0.0
+      ke_c_tgt = 0.0
+      do k=1,nz
+        ke_c_src = ke_c_src + h1(k) * (v_src(k) - v_bt)**2
+        ke_c_tgt = ke_c_tgt + h2(k) * (v_tgt(k) - v_bt)**2
+      enddo
+      ! Next rescale baroclinic component on target grid to conserve ke
+      if (ke_c_src < 1.5625 * ke_c_tgt) then
```

Comment:
Please add a comment explaining where 1.25 and 1.5625 = 1.25**2 come from, and whether these are arbitrary hard-coded numbers or whether this value could reasonably chosen to be different.

Reply:
I still think that an explanation in a comment of where 1.25 and 1.5625 come from would be valuable, but I do not want to hold up this PR waiting for this clarifying comment.

Reply:
I did add the comment - not sure how to resolve this conversation though.

https://github.com/NCAR/MOM6/blob/982b5776a51bed07f9d811092c4b8b8fbb65960e/src/ALE/MOM_ALE.F90#L1176-L1180